src/helpers.ts

```diff
@@ -43,8 +47,9 @@ const nextRouterAction = async ({
 };
 
 const routerPush = async (href: string) => {
-  if (store.getState().adapter === 'nextjs') {
-    await nextRouterAction({ href });
+  const { adapter, replace } = store.getState();
+  if (adapter === 'nextjs') {
+    await nextRouterAction({ href, replace });
   } else {
     window.history.pushState({ path: href }, '', href);
```

Comment:
I think we should also handle the replace here. We can do `history.replace` I believe